### ISSUE TITLE
`pypesto.profile.parameter_profile` leaves `Problem` in modified state if an error occurs during profiling

### ISSUE THREAD

**Issue by dweindl** [ICB-DCM/pyPESTO#1150]
What happens:

```python
fixed_parameters_before = problem.x_fixed_indices

parameter_profile(problem, profile_index=[0], ...) # <- fails for unrelated reasons

assert problem.x_fixed_indices == fixed_parameters_before # <- fails, because parameters were fixed during profiling but not unfixed

# retry computing profiles
parameter_profile(problem, profile_index=[0], ...) # <- nothing happens, because that parameter is still set to fixed
```

`parameter_profile` should either create a copy of `problem`, or make sure that it is in the original state when `parameter_profile` exists, independent of any potential errors.

**Comment by PaulJonasJost** [comment]
Would go with copy for now. Any strong arguments against that?

**Comment by dweindl** [comment]
> Would go with copy for now. Any strong arguments against that?

No. Considering the cost of profiling, the any potential copying-overhead should be negligible. 
That Problem/Objective is copy-able is already expected elsewhere, so it shouldn't be a problem here either.

**Comment by PaulJonasJost** [comment]
yes, that's what I thought as well. 👍🏼

**Comment by dweindl** [comment]
Somewhat independent issue, but related to the pseudo-code above: Would be good to issue some warning when attempting to compute profiles for fixed parameters instead of just not doing anything.

**Comment by PaulJonasJost** [comment]
You are right, just went through the profile code and noticed that as well. As it Is a somewhat independent issue, I will create a new issue and fix it as well 👍🏼
